api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -838,3 +838,17 @@ message RequestIDExtension {
   // Request ID extension specific configuration.
   google.protobuf.Any typed_config = 1;
 }
+
+// Match input indicates that matching should be done on a specific request header.
+// TODO(snowp): Link to unified matching docs.
+message HttpRequestHeaderMatchInput {
```

Comment:
Will this also be needed in the route table?  If so, should we put this somewhere more generic, so that it can be used in both places?

Reply:
I was imagining that we wouldn't depend on this directly in the route table if we ended up using the matching API for it? Wouldn't we just refer to this via opaque TypedExtensionConfigs?

Any suggestions for where this would fit better? I can dump it into the extensions/common/matcher package I put the ExtensionWithMatcher, but there might be a better location for it

Reply:
Yes, we'd use `TypedExtensionConfig` to plug this into the matcher proto.  But if we're matching against HTTP headers in both cases, then we'd presumably want to use the same proto to plugin to the `TypedExtensionConfig` in both the filter matcher case and the route table case.

Maybe a better location would be type/matcher/http_header.proto?